src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -121,6 +121,14 @@
     <Field Name="IsNewInstance" Type="bool" Null="NotApplicable"/>
   </Node>
 
+  <!--
+  This node is used to represent an expression of a certain type, when attempting to bind its Index or Range pattern indexer
+  It does not survive past initial binding.
```

Comment:
You can add a case to `LocalRewritingValidator` to enforce that.

Reply:
Actually it will get past binding but will be replaced with another node in the rewriter.
Regardless, I think the visitor is unable to reach it because it's captured in `IndexerArgumentInfo.ArgumentsOpt` which is not a bound node.

Reply:
nit: May be worth adjusting the comment to reflect

Reply:
(pending)